SSA/Projects/InstCombine/HackersDelight.lean

```diff
@@ -164,16 +169,88 @@ theorem neq_iff_sub_or_sub :
     x ≠ y ↔ (x - y ||| y - x).getMsb 0 := by
   sorry
 
+theorem neq_iff_neg_sub_abs :
+    x ≠ y ↔ (-(x - y).abs).getMsb 0 := by
+  sorry
+
 theorem lt_iff_sub_xor_xor_and_sub_xor :
-    x < y ↔ ((x - y) ^^^ ((x ^^^ y) &&& ((x - y) ^^^ x))).getMsb 0 := by
+    (x <ₛ y) ↔ ((x - y) ^^^ ((x ^^^ y) &&& ((x - y) ^^^ x))).getMsb 0 := by
   sorry
 
 theorem lt_iff_and_not_or_not_xor_and_sub :
-    x < y ↔ ((x &&& ~~~ y) ||| (~~~ (x ^^^ y) &&& (x - y))).getMsb 0 := by
+    (x <ₛ y) ↔ ((x &&& ~~~ y) ||| (~~~ (x ^^^ y) &&& (x - y))).getMsb 0 := by
   sorry
 
 theorem le_iff_or_not_and_xor_or_not_sub :
-    x ≤ y ↔ ((x ||| ~~~ y) &&& ((x ^^^ y) ||| ~~~ (y - x))).getMsb 0 := by
+    (x ≤ₛ y) ↔ ((x ||| ~~~ y) &&& ((x ^^^ y) ||| ~~~ (y - x))).getMsb 0 := by
+  sorry
+
+theorem ult_iff_not_and_or_not_xor_and_sub :
+    (x <ᵤ y) ↔ ((~~~ x &&& y) ||| (~~~ (x ^^^ y) &&& (x - y))).getMsb 0 := by
+  sorry
+
+theorem ule_iff_not_or_and_xor_or_not_sub :
+    (x ≤ᵤ y) ↔ ((~~~ x ||| y) &&& ((x ^^^ y) ||| ~~~ (y - x))).getMsb 0 := by
+  sorry
+
+theorem eq_zero_iff_abs_sub :
+    x = 0 ↔ (x.abs - 1).getMsb 0 := by
+  sorry
+
+theorem eq_zero_iff_not_or_sub :
+    x = 0 ↔ (~~~ (x ||| -x)).getMsb 0 := by
+  sorry
+
+theorem eq_zero_iff_not_and_sub :
+    x = 0 ↔ (~~~ x &&& x - 1).getMsb 0 := by
+  sorry
+
+theorem neq_zero_iff_or_neg :
+    x ≠ 0 ↔ (x ||| -x).getMsb 0 := by
+  sorry
+
+theorem neq_zero_iff_neg_abs :
+    x ≠ 0 ↔ (-(x.abs)).getMsb 0 := by
+  sorry
+
+theorem lt_zero_iff :
```

Comment:
In keeping with the above convention, I suggest renameing this to "slt_zero_iff"

```suggestion
theorem slt_zero_iff :
```

Reply:
ah yes, I started doing this with the unsigned inequalities but forgot these, I will update in upcoming commit